Code/Tools/ProjectManager/Source/AdjustableHeaderWidget.cpp

```diff
@@ -0,0 +1,120 @@
+        : QTableWidget(parent)
+    {
+        setObjectName("adjustableHeaderWidget");
+        setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Minimum);
+        setFixedHeight(s_headerWidgetHeight);
+
+        m_header = horizontalHeader();
+        m_header->setDefaultAlignment(Qt::AlignLeft);
+
+        setColumnCount(headerLabels.count());
+        setHorizontalHeaderLabels(headerLabels);
+
+        AZ_Assert(defaultHeaderWidths.count() == columnCount(), "Default header widths does not match number of columns");
+        AZ_Assert(resizeModes.count() == columnCount(), "Resize modesdoes not match number of columns");
+
+        for (int column = 0; column < columnCount(); ++column)
+        {
+            m_header->resizeSection(column, defaultHeaderWidths[column]);
+            m_header->setSectionResizeMode(column, resizeModes[column]);
+        }
+
+        m_header->setMinimumSectionSize(minHeaderWidth);
+        m_header->setCascadingSectionResizes(true);
+
+        connect(m_header, &QHeaderView::sectionResized, this, &AdjustableHeaderWidget::OnSectionResized);
+    }
+
+    void AdjustableHeaderWidget::OnSectionResized(int logicalIndex, int oldSize, int newSize)
+    {
+        const int headerCount = columnCount(); 
+        const int headerWidth = m_header->width();
+        const int totalSectionWidth = m_header->length();
+
+        if (totalSectionWidth > headerWidth && newSize > oldSize)
+        {
+            int xPos = 0;
+            int requiredWidth = 0;
+
+            for (int i = 0; i < headerCount; i++)
+            {
+                if (i < logicalIndex)
+                {
+                    xPos += m_header->sectionSize(i);
+                }
+                else if (i == logicalIndex)
+                {
+                    xPos += newSize;
+                }
+                else if (i > logicalIndex)
+                {
+                    if (m_header->sectionResizeMode(i) == QHeaderView::ResizeMode::Fixed)
+                    {
+                        requiredWidth += m_header->sectionSize(i);
+                    }
+                    else
+                    {
+                        requiredWidth += m_header->minimumSectionSize();
+                    }
+                }
+            }
+
+            if (xPos + requiredWidth > headerWidth)
+            {
+                m_header->resizeSection(logicalIndex, oldSize);
+            }
+        }
+
+        // wait till all columns resized
+        QTimer::singleShot(0, [&]()
+        {
+            // only re-paint when the header and section widths have settled
+            const int headerWidth = m_header->width();
+            const int totalSectionWidth = m_header->length();
+            if (totalSectionWidth == headerWidth)
+            {
+                emit sectionsResized();
+            }
+        });
+    }
+
+    int AdjustableHeaderWidget::CalcHeaderXPos(int headerIndex, bool calcEnd) const
```

Comment:
Maybe consider changing this to something along the lines of `CalcColumnXBounds` retuning the left and right positions as a pair instead having the optional toggle.  It would improve clarity of usage.